tests/test_Freecell.py

```diff
@@ -0,0 +1,136 @@
+from unittest import TestCase
+from Freecell import FreeCell
+from Game import State
+from Card import Card
+
+
+class TestFreecell(TestCase):
+
+    def __flatten_list(self, _list: list) -> list:
+        return [elem for sublist in _list for elem in sublist]
+
+    def setup_scenario_various_moves(self) -> FreeCell:
+        return FreeCell()
+
+    def setup_scenario_stack_move(self) -> FreeCell:
+        freecell = FreeCell()
+        freecell.board.columns = [
+            [Card(13, 'h')], [Card(13, 'd')], [Card(13, 's')], [Card(13, 'c')],
+            [Card(4, 'h')], [Card(4, 'd'), Card(2, 'c')], [Card(4, 's')], [Card(4, 'c')]
+        ]
+        freecell.board.free_cells = [
+            Card(10, 'h'), Card(10, 'd'), Card(10, 's'), Card(10, 'c')
+        ]
+        freecell.board.suit_stack = {
+            'h': Card(1, 'h'),
+            'd': Card(1, 'd'),
+            's': Card(1, 's'),
+            'c': Card(1, 'c')
+        }
+        return freecell
+
+    def setup_scenario_free_column(self) -> FreeCell:
+        freecell = FreeCell()
+        freecell.board.columns = [
+            [], [Card(13, 'd'), Card(7, 'c')], [Card(13, 's')], [Card(13, 'c')],
+            [Card(3, 'h')], [Card(3, 'd')], [Card(3, 's')], [Card(3, 'c')]
+        ]
+        freecell.board.free_cells = [
+            Card(10, 'h'), Card(10, 'd'), Card(10, 's'), Card(10, 'c')
+        ]
+        freecell.board.suit_stack = {
+            'h': Card(1, 'h'),
+            'd': Card(1, 'd'),
+            's': Card(1, 's'),
+            'c': Card(1, 'c')
+        }
+        return freecell
+
+    def setup_scenario_empty_board(self) -> FreeCell:
+        freecell = FreeCell()
+        freecell.board.columns = [[], [], [], [], [], [], [], []]
+        freecell.board.free_cells = [None, None, None, None]
+        freecell.board.suit_stack = {
+            'h': Card(13, 'h'),
+            'd': Card(13, 'd'),
+            's': Card(13, 's'),
+            'c': Card(13, 'c')
+        }
+        return freecell
+
+    def setup_scenario_no_moves(self) -> FreeCell:
```

Comment:
Moze warto jedna metode setup zrobic bardziej w sposob integracyjny? czyli uzywajac metod z board.

Np. zamiast ustawiac wszystko na sztywno =.
Mozemy uzyc konstruktora z pusta lista jako parametr i osiagniemy efekt 
freecell.board.columns = [[], [], [], [], [], [], [], []]
taki bardziej rozbudowany.

Nie chodzi mi ze ta metoda jest zla bo jest dobra i widac od razu jakie sa zalozenia - tak powinno to wygladac. Mowie juz nie o testach jednostkowych ale bardziej o takim rozbudowanym jednym chociaz zeby zobaczyc czy wszystko dziala tak jak chcemy - symulacja gry taka ala.
Co myślisz??